anesthetic/plotting/_matplotlib/core.py

```diff
@@ -0,0 +1,45 @@
+from pandas.plotting._matplotlib.core import (ScatterPlot as _ScatterPlot,
+                                              HexBinPlot as _HexBinPlot,
+                                              MPLPlot)
+from anesthetic.weighted_pandas import _WeightedObject
+
+
+def _get_weights(kwargs, data):
+    if isinstance(data, _WeightedObject):
+        kwargs['weights'] = data.weights
+
+
+class _WeightedMPLPlot(MPLPlot):
+    def __init__(self, data, *args, **kwargs):
+        _get_weights(kwargs, data)
+        super().__init__(data, *args, **kwargs)
+
+
+def _compress_weights(kwargs, data):
+    if isinstance(data, _WeightedObject):
+        return data.compress(kwargs.pop('ncompress', None))
+    else:
+        return data
+
+
+class _CompressedMPLPlot(MPLPlot):
+    def __init__(self, data, *args, **kwargs):
+        data = _compress_weights(kwargs, data)
+        super().__init__(data, *args, **kwargs)
```

Comment:
This automates compression of weighted samples into unweighted samples for scatter and bootstrap plots.

Reply:
Do we have a test for this?

Reply:
No, but it is covered by anything that derives from it (e.g. `ScatterPLot`). Since it isn't a public function, I wouldn't consider this warranting a test, since in theory the underlying implementation could change without changing the public behaviour (and underscore prefixes are meant to indicate this to users).

Reply:
My question was more regarding your comment rather than this specific class/method, i.e. whether the compressed samples match the expected unweighted samples. But I guess this is automatically tested by testing scatter and bootstrap plots...?